### ISSUE TITLE
channels for python 2.7 broken?

### ISSUE THREAD

**Issue by DebVortex** [andrewgodwin/channels#202]
Hi there,

I'm not sure if this Issue is correct here or If it should rather belong to [asgiref](https://github.com/andrewgodwin/asgiref/).

it seems that ``asgiref==0.13.3`` does not play well with ``channels==0.14.0``.

The ``valid_channel_name`` function checks that the name of the channel is a proper ``six.text_type`` (which is unicode in python27). However the channel names, generated from channels are ``str`` not ``unicode``. Python34 works fine as expected.

**Comment by andrewgodwin** [comment]
Interesting - channel names are meant to be unicode, so this is likely a bug in Channels surfaced by the more stringent check. I'll look into it later - what exactly is the stack trace you're getting looking like?

**Comment by DebVortex** [comment]
Here is a short exmaple stack trace. 

```
__________________________________________________________ 
TestWebsocketConsumer.test_ws_disconnect 
___________________________________________________________
tests/core/test_consumers.py:58: in test_ws_disconnect
    client.receive()  # Drop connected message
.tox/py27/local/lib/python2.7/site-packages/channels/tests/base.py:124: in receive
    message = self.get_next_message(self.reply_channel)
.tox/py27/local/lib/python2.7/site-packages/channels/tests/base.py:90: in get_next_message
    recv_channel, content = channel_layers[self.alias].receive_many([channel])
.tox/py27/local/lib/python2.7/site-packages/asgiref/inmemory.py:58: in receive_many
    assert all(self.valid_channel_name(channel) for channel in channels), "One or more channel names invalid"
E   AssertionError: One or more channel names invalid
```

Looking at it... could it be a problem in the test client?

**Comment by Krukov** [comment]
Yeah  problem in the test client, it's my bad. :suspect:

**Comment by andrewgodwin** [comment]
@DebVortex If you could confirm latest master works fine for you I'll release it as a patch release later on

**Comment by DebVortex** [comment]
Master works perfectly fine! Thanks.

**Comment by andrewgodwin** [comment]
Fantastic, released in `0.14.1`.
